### ISSUE TITLE
CA1062 False positive when value is overwritten

### ISSUE THREAD

**Issue by bexxx** [dotnet/roslyn-analyzers#2339]
#### Analyzer package
Microsoft.CodeAnalysis.FxCopAnalyzers

#### Package Version
Example: v2.9.1

#### Diagnostic ID
Error	CA1062	In externally visible method 'string ExceptionExtensions.AsString(Exception exception)', validate parameter 'exception' is non-null before using it. If appropriate, throw an ArgumentNullException when the argument is null or add a Code Contract precondition asserting non-null argument.	Test.Common (net461)	

#### Repro steps
```csharp
        public static string AsString(this Exception exception)
        {
            if (exception == null)
            {
                throw new ArgumentNullException(nameof(exception));
            }

            while (exception.InnerException != null)
            {
                exception = exception.InnerException;
            }

            return "...";
        }
```

#### Expected behavior
No error
#### Actual behavior
Error CA1062 for line with while

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/2345
